### PR TITLE
Recipe fragment

### PR DESCRIPTION
**Pull request by Calvin906**
I accidentally had some files in here that should have gone into the drawer branch but oh well. The recipe fragment is up and pulls from pastebin

### CODE REVIEW

File: app/src/main/java/preston/com/recipeapp/DrawerRVAdapter.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+
+    }
+
+    @Override
+    public int getItemCount() {
+        return 0;
+    }
+
+    private class DrawerImageViewHolder extends RecyclerView.ViewHolder {
```
Comment (JPrez38) [inline app/src/main/java/preston/com/recipeapp/DrawerRVAdapter.java]: this image isn't going to be in the list. It should be separate and apart of the xml. The only reason the recipe image was in the list is so you can scroll it

File: app/src/main/java/preston/com/recipeapp/MyAsyncTaskLoaderRecipe.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package preston.com.recipeapp;
+
+import android.content.AsyncTaskLoader;
+import android.content.Context;
+
+/**
+ * Created by Alex Preston on 1/27/16.
+ */
+public abstract class MyAsyncTaskLoaderRecipe<T> extends AsyncTaskLoader<T> {
```
Comment (JPrez38) [inline app/src/main/java/preston/com/recipeapp/MyAsyncTaskLoaderRecipe.java]: probably doesn't need recipe in the name. Just CustomAsyncLoader to keep it generic so you can use this for non-recipe stuff.

File: app/src/main/java/preston/com/recipeapp/RecipeFragment.java [hunk 3]
```diff
@@ -1,9 +1,154 @@
...
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * Created by Alex Preston on 1/5/16.
  */
-public class RecipeFragment extends AppCompatActivity {
+public class RecipeFragment extends AppCompatActivity implements LoaderManager.LoaderCallbacks<List<Recipe>> {
```
Comment (JPrez38) [inline app/src/main/java/preston/com/recipeapp/RecipeFragment.java]: huh? you called this class fragment but it extends from appcompatactivity

Comment (Calvin906) [inline app/src/main/java/preston/com/recipeapp/RecipeFragment.java]: I tried to make it a fragment. Should it be a Activity? I understand now why it is wrong

File: app/src/main/java/preston/com/recipeapp/RecipeFragment.java [hunk 4]
```diff
@@ -1,9 +1,154 @@
...
+
+                return download("http://pastebin.com/raw/TiLUEnyx");
+            }
+        };
+    }
+
+    @Override
+    public void onLoadFinished(Loader<List<Recipe>> loader, List<Recipe> data) {
+        for (int i = 0; i < data.size(); i++) {
+            mRVAdapter.setData(data.get(i).getRecipeName(), data.get(i).getSteps(), data.get(i).getIngredients(), data.get(i).getImage());
```
Comment (JPrez38) [inline app/src/main/java/preston/com/recipeapp/RecipeFragment.java]: Don't make your set data method like this. Just make it setData(List<Recipe> data) and pull the actual items out in the adapter itself.

File: app/src/main/java/preston/com/recipeapp/RecipeFragment.java [hunk 5]
```diff
@@ -1,9 +1,154 @@
...
+
+    }
+
+    public List<Recipe> download(String website) {
+        List<Recipe> recipes = new ArrayList<>();
+        String toOpen = website;
+
+        URL pasteBin = null;
+
+        try {
```
Comment (JPrez38) [inline app/src/main/java/preston/com/recipeapp/RecipeFragment.java]: These can all be wrapped in a single try catch block. One try with several catch statements. It makes this method look much bigger than it needs to be.

File: app/src/main/java/preston/com/recipeapp/RecipeRVAdapter.java [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+import com.squareup.picasso.Picasso;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Alex Preston on 1/7/16.
+ */
+public class RecipeRVAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
```
Comment (JPrez38) [inline app/src/main/java/preston/com/recipeapp/RecipeRVAdapter.java]: What's an RV adapter? haha. Sounds like an adapter for RVs. Just call it RecipeAdapter

File: app/src/main/java/preston/com/recipeapp/RecipeRVAdapter.java [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        View view;
+
+        switch (viewType) {
+            case IMAGE:
+                view = inflater.inflate(R.layout.image_view, parent, false);
+                return new RecipeViewHolderImage(view);
```
Comment (JPrez38) [inline app/src/main/java/preston/com/recipeapp/RecipeRVAdapter.java]: nice

**Comment by JPrez38** [comment]
looking pretty good in the hood. Make the above changes and you can merge it. I image you still have xml issues, but that you're still working on them. But looks good bro. Well done!

**Pull request merged**